mxcube3/routes/Login.py

```diff
@@ -93,84 +93,84 @@ def get_initial_state():
         :statuscode: 200: no error
         :statuscode: 409: error occurred
     """
-    motors = ['Phi', 'Focus', 'PhiZ', 'PhiY', 'Zoom', 'BackLightSwitch','BackLight','FrontLightSwitch', 'FrontLight','Sampx', 'Sampy'] 
+    motors = ['Phi', 'Focus', 'PhiZ', 'PhiY','Sampx', 'Sampy'] 
     #'Kappa', 'Kappa_phi',
     data = {}
     data['Motors'] = {}
+    for mot in motors:
+       motor_hwobj = mxcube.diffractometer.getObjectByRole(mot.lower())
+       if motor_hwobj is not None:
+           try:
+               pos = motor_hwobj.getPosition()
+               status = motor_hwobj.getState()
+           except Exception:
+               logging.getLogger('HWR').exception('[SAMPLEVIEW] could not get "%s" motor' %mot)
+           data['Motors'].update({mot: {'Status': status, 'position': pos}})
+
     try:
-        for mot in motors:
-            motor_hwobj = mxcube.diffractometer.getObjectByRole(mot.lower())
-            if motor_hwobj is not None:
-                if mot == 'Zoom':
-                    pos = motor_hwobj.predefinedPositions[motor_hwobj.getCurrentPositionName()]
-                    status = "unknown"
-                elif mot == 'BackLightSwitch' or mot == 'FrontLightSwitch':
-                    states = {"in": 1, "out": 0}
-                    pos = states[motor_hwobj.getActuatorState()]  # {0:"out", 1:"in", True:"in", False:"out"}
-                    # 'in', 'out'
-                    status = pos 
-                else:
-                    try:
-                        pos = motor_hwobj.getPosition()
-                        status = motor_hwobj.getState()
-                    except Exception:
-                        logging.getLogger('HWR').exception('[SAMPLEVIEW] could not get "%s" motor' %mot)
-                data['Motors'].update({mot: {'Status': status, 'position': pos}})
-        data['Camera'] = {'pixelsPerMm': mxcube.diffractometer.get_pixels_per_mm(),
-            'imageWidth':  mxcube.diffractometer.image_width,
-            'imageHeight':  mxcube.diffractometer.image_height,
-            }
+        motor_hwobj = mxcube.diffractometer.getObjectByRole('zoom')
+        data['Motors'].update({"Zoom": {"Status":motor_hwobj.getState(), "position": motor_hwobj.predefinedPositions[motor_hwobj.getCurrentPositionName()] }})
+    except Exception:
+        logging.getLogger('HWR').exception('[SAMPLEVIEW] could not get Zoom motor')
+    
+    for light in ('BackLight','FrontLight'):
+        hwobj = mxcube.diffractometer.getObjectByRole(light)
+        switch_state = 1 if hwobj.getActuatorState()=='in' else 0
```

Comment:
We should agree on which hwobj to use. Personally I find it good to separate motors
and things like backlight and frontlight. Zoom is also a bit different so that's why we excluded it
from the Motors list.
So, which hwobj are you using for the lights?
The lights are both an actuator (in/out) and a "motor" (with an intensity value).